lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -169,15 +178,14 @@ def ldap_search(self, email, username, options):
                             # keep role names as list
                             params[self.role_search_option] = attrs[attr]
                         elif attr in attrs:
-                            params[attr] = str(attrs[attr][0])
+                            params[attr] = ' '.join(attrs[attr])
```

Comment:
Wouldn't it be better to keep `params['memberOf']` as an array? Spaces can be present in the entries.
That could be implemented by restoring this line and changing line 177 to:
```python
                        if attr == 'memberOf' or (self.role_search_attribute and attr == self.role_search_attribute[1:-1]):
```

Reply:
As such, it crash (KeyError exception) when trying to access params['memberOf'] at line 219 : `if search_filter not in params['memberOf']:` . Likely because `self.role_search_option` does'nt have any value at that point. We have dedicate one new 'switch' case to this. In addition since its now a list and we are doing a partial search on it we have to use list comprehension to handle it wereas it was simple 'in' string check in a string before.

Reply:
Why do you need a partial search, by the way?

Reply:
In ours case, we are not responsible of the AD node creation, so we can't choose their names. The internal AD naming rules leads to names like '6452_GALAXY_TEST' when we typically want to use matching term such as 'GALAXY_TEST' or eventually 'GALAXY'.  We also use full names (exact match thus), for other galaxy instances.

Reply:
So `6452_GALAXY_TEST` is the group name (not username)? Weird!